Physics/StreamerInception/CD_StreamerInceptionStepperImplem.H

```diff
@@ -1052,11 +1057,27 @@ StreamerInceptionStepper<P, F, C>::computeDt()
       break;
     }
     }
-
+    auto computeGradient = [this](const Real& a_timeStep) {
+      return (m_voltageCurve(m_time + a_timeStep) - m_voltageCurve(m_time)) / a_timeStep;
+    };
+    if(m_prevDt == 0){
+      m_prevDt = m_minDt;
+    }
+    Real       currGradient = computeGradient(m_prevDt);
+    while(currGradient == 0){
+      m_prevDt += m_minDt;
+      currGradient = computeGradient(m_prevDt);
+    }
+    Real currVoltage  = m_voltageCurve(m_time);
+    if(currVoltage == 0){
+      currVoltage = m_voltageCurve(m_time+m_prevDt);
+    }
+    dt = std::abs(m_deltaVPercent*currVoltage/currGradient);
+    
     dt = std::min(dt, m_maxDt);
     dt = std::max(dt, m_minDt);
   }
-
+  m_prevDt = dt;
```

Comment:
Set m_prevDt = 0.0 in constructor for failsafe. 

The advance method does not _have_ to use the dt that is computed here, so m_prevDt should be set to a_dt in the advance method

Reply:
m_prevDt is set to 0 in the header-file (i.e. initialized to 0 at construction) already. I can change that to 0.0

Ah right, I'll change that now

Reply:
Or I can change it so it is initialized in the initializer list instead? Up to you